libsplinter/src/admin/service/shared.rs

```diff
@@ -5830,6 +6101,473 @@ mod tests {
+            peer_connector,
+            store,
+            signature_verifier,
+            Box::new(MockAdminKeyVerifier::default()),
+            Box::new(AllowAllKeyPermissionManager),
+            writer,
+            #[cfg(feature = "admin-service-event-store")]
+            event_store,
+        );
+
+        // Add the circuit to be abandoned
+        admin_shared
+            .admin_store
+            .add_circuit(
+                store_circuit(CIRCUIT_PROTOCOL_VERSION, StoreCircuitStatus::Active),
+                store_circuit_nodes(),
+            )
+            .expect("unable to add circuit to store");
+
+        if let Ok(()) = admin_shared.validate_abandon_circuit("01234-ABCDE", PUB_KEY, "node_a", 1) {
+            panic!("Should have been invalid because the protocol version is invalid");
+        }
+
+        shutdown(mesh, cm, pm);
+    }
+
+    #[cfg(feature = "circuit-abandon")]
+    /// Tests that a request to abandon a circuit returns an error if the requester is not
+    /// permitted for the admin service.
+    ///
+    /// 1. Set up `AdminServiceShared`
+    /// 2. Add a `Active` circuit to the admin store
+    /// 3. Call `validate_abandon_circuit` with the circuit and requester info
+    /// 4. Validate the call to `validate_abandon_circuit` returns an error
+    #[test]
+    fn test_validate_abandon_circuit_not_permitted() {
+        let store = setup_admin_service_store();
+        #[cfg(feature = "admin-service-event-store")]
+        let event_store = store.clone_boxed();
+        let (mesh, cm, pm, peer_connector) = setup_peer_connector(None);
+        let orchestrator = setup_orchestrator();
+
+        let signature_verifier = Secp256k1Context::new().new_verifier();
+
+        let table = RoutingTable::default();
+        let writer: Box<dyn RoutingTableWriter> = Box::new(table.clone());
+
+        let admin_shared = AdminServiceShared::new(
+            "node_a".into(),
+            Arc::new(Mutex::new(orchestrator)),
+            #[cfg(feature = "service-arg-validation")]
+            HashMap::new(),
+            peer_connector,
+            store,
+            signature_verifier,
+            Box::new(MockAdminKeyVerifier::new(false)),
+            Box::new(AllowAllKeyPermissionManager),
+            writer,
+            #[cfg(feature = "admin-service-event-store")]
+            event_store,
+        );
+
+        // Add the circuit to be abandoned
+        admin_shared
+            .admin_store
+            .add_circuit(
+                store_circuit(CIRCUIT_PROTOCOL_VERSION, StoreCircuitStatus::Active),
+                store_circuit_nodes(),
+            )
+            .expect("unable to add circuit to store");
+
+        if let Ok(()) = admin_shared.validate_abandon_circuit(
+            "01234-ABCDE",
+            PUB_KEY,
+            "node_a",
+            ADMIN_SERVICE_PROTOCOL_VERSION,
+        ) {
+            panic!("Should have been invalid because the protocol version is invalid");
```

Comment:
based on the description of the test this message should probably be something like 
`"Should have been invalid because the requester is not permitted for the admin service"`

Reply:
Fixed